packages/core/src/util/file-type.ts

```diff
@@ -126,26 +147,26 @@ export function detectFileType(document: string): [string, string] {
     fileBuffer[4] === XML_MAGIC_NUMBER_5 &&
     fileBuffer[5] === XML_MAGIC_NUMBER_6
   ) {
-    return [XML_APP_MIME_TYPE, XML_FILE_EXTENSION];
+    return { mimeType: XML_APP_MIME_TYPE, fileExtension: XML_FILE_EXTENSION };
   } else if (
     fileBuffer[0] === PNG_MAGIC_NUMBER_1 &&
     fileBuffer[1] === PNG_MAGIC_NUMBER_2 &&
     fileBuffer[2] === PNG_MAGIC_NUMBER_3 &&
     fileBuffer[3] === PNG_MAGIC_NUMBER_4
   ) {
-    return [PNG_MIME_TYPE, PNG_FILE_EXTENSION];
+    return { mimeType: PNG_MIME_TYPE, fileExtension: PNG_FILE_EXTENSION };
   } else if (
     fileBuffer[0] === JPEG_MAGIC_NUMBER_1 &&
     fileBuffer[1] === JPEG_MAGIC_NUMBER_2 &&
     fileBuffer[2] === JPEG_MAGIC_NUMBER_1
   ) {
-    return [JPEG_MIME_TYPE, JPEG_FILE_EXTENSION];
+    return { mimeType: JPEG_MIME_TYPE, fileExtension: JPEG_FILE_EXTENSION };
   } else if (fileBuffer[0] === BMP_MAGIC_NUMBER_1 && fileBuffer[1] === BMP_MAGIC_NUMBER_2) {
-    return [BMP_MIME_TYPE, BMP_FILE_EXTENSION];
-  } else if (isLikelyTextFile(Buffer.from(document))) {
-    return [TXT_MIME_TYPE, TXT_FILE_EXTENSION];
+    return { mimeType: BMP_MIME_TYPE, fileExtension: BMP_FILE_EXTENSION };
+  } else if (isLikelyTextFile(fileBuffer)) {
```

Comment:
bug: isLikelyTextFile is getting called on only 6 bytes if detectFileTypeBuf is called by detectFileTyp